eth/core/precompile/method.go

```diff
@@ -80,13 +74,64 @@ type Method struct {
 // ValidateBasic returns an error if this a precompile `Method` has invalid fields.
 func (m *Method) ValidateBasic() error {
 	// ensure all required fields are nonempty
-	if len(m.AbiSig) == 0 || m.AbiMethod != nil || m.Execute == nil {
+	if len(m.AbiSig) == 0 || !m.Execute.IsValid() || m.Execute.IsNil() {
 		return ErrIncompleteMethod
 	}
 
 	return nil
 }
 
+// Call executes the precompile's executable with the given context and input arguments.
+func (m *Method) Call(ctx []reflect.Value, input []byte) ([]byte, error) {
+	// Unpack the args from the input, if any exist.
+	unpackedArgs, err := m.AbiMethod.Inputs.Unpack(input[NumBytesMethodID:])
+	if err != nil {
+		return nil, err
+	}
+
+	// Build argument list
+	reflectedUnpackedArgs := make([]reflect.Value, 0, len(unpackedArgs))
+	for _, unpacked := range unpackedArgs {
+		reflectedUnpackedArgs = append(reflectedUnpackedArgs, reflect.ValueOf(unpacked))
+	}
+
+	// Call the executable
+	results := m.Execute.Call(append(ctx, reflectedUnpackedArgs...))
+
+	// If the precompile returns something wrong (e.g. wrong number of results), the error is
+	// returned to the caller.
+	if len(results) != 2 { //nolint:gomnd // its okay.
+		return nil, errors.Wrapf(
+			vm.ErrExecutionReverted,
+			"vm error: precompile [%s] returned [%d] results, expected 2",
+			debug.GetFnName(m.Execute.Interface()), len(results),
+		)
+	}
```

Comment:
this should not happen here. at registration you should statically verify that there are only 2 return values on that function

Reply:
results is after the actual call is done, this doesn't have to do with the return values of the function. the two is the first return value and the second is possible errors just the way that reflect works

Reply:
yeah @calbera I think this is more a reflect thing, although @shampoobera idk if we need it, may be a redudant check (although always better to check more than less)